src/da.rs

```diff
@@ -51,6 +57,8 @@ pub struct CelestiaConnection {
     rx: Arc<tokio::sync::Mutex<mpsc::Receiver<Message>>>,
 }
 
+pub struct InMemoryDataAvailabilityLayer {}
```

Comment:
I would maybe rename to LocalDataAvailabilityLayer. Also please add a comment describing what it is and that its only for testing. Could we put it behind a testing feature?